Lib/test/test_concurrent_futures/test_process_pool.py

```diff
@@ -218,6 +233,81 @@ def mock_start_new_thread(func, *args, **kwargs):
+            executor._terminate_or_kill_workers = unittest.mock.Mock()
+            executor.kill_workers()
+
+        executor._terminate_or_kill_workers.assert_called_once_with(operation=futures.process._KILL)
+
+    def test_process_pool_executor_terminate_or_kill_workers_invalid_op(self):
+        with futures.ProcessPoolExecutor(max_workers=1) as executor:
+            self.assertRaises(ValueError,
+                              executor._terminate_or_kill_workers,
+                              operation='invalid operation'),
+
+    @parameterize(*TERMINATE_OR_KILL_PARAMS)
+    def test_process_pool_executor_terminate_kill_workers(self, function_name):
+        manager = self.get_context().Manager()
+        q = manager.Queue()
+
+        with futures.ProcessPoolExecutor(max_workers=1) as executor:
+            executor.submit(_put_sleep_put, q)
+
+            # We should get started, but not finished since we'll terminate the
+            # workers just after
+            self.assertEqual(q.get(timeout=5), 'started')
+
+            worker_process = list(executor._processes.values())[0]
+            getattr(executor, function_name)()
+            worker_process.join()
+
+            if function_name == 'terminate_workers' or \
+                sys.platform == 'win32':
+                # On windows, kill and terminate both send SIGTERM
+                self.assertEqual(worker_process.exitcode, -signal.SIGTERM)
+            elif function_name == 'kill_workers':
+                self.assertEqual(worker_process.exitcode, -signal.SIGKILL)
+
+            self.assertRaises(queue.Empty, q.get, timeout=1)
+
+    @parameterize(*TERMINATE_OR_KILL_PARAMS)
+    def test_process_pool_executor_terminate_kill_workers_dead_workers(self, function_name):
+        with futures.ProcessPoolExecutor(max_workers=1) as executor:
+            future = executor.submit(os._exit, 1)
+            self.assertRaises(BrokenProcessPool, future.result)
+
+            # even though the pool is broken, this shouldn't raise
+            getattr(executor, function_name)()
+
+    @parameterize(*TERMINATE_OR_KILL_PARAMS)
+    def test_process_pool_executor_terminate_kill_workers_not_started_yet(self, function_name):
+        ctx = self.get_context()
+        with unittest.mock.patch.object(ctx, 'Process') as mock_process:
+            with futures.ProcessPoolExecutor(max_workers=1, mp_context=ctx) as executor:
+                # The worker has not been started yet, terminate/kill_workers
+                # should basically no-op
+                getattr(executor, function_name)()
+
+            mock_process.return_value.kill.assert_not_called()
+            mock_process.return_value.terminate.assert_not_called()
+
+    @parameterize(*TERMINATE_OR_KILL_PARAMS)
+    def test_process_pool_executor_terminate_kill_workers_stops_pool(self, function_name):
+        with futures.ProcessPoolExecutor(max_workers=1) as executor:
+            executor.submit(time.sleep, 0).result()
```

Comment:
```suggestion
            task = executor.submit(time.sleep, 0)
            task.result()
```

I prefer having the calls split in case of an error. Not sure however if `task.result()` should return None.